fabric-armor-api-v1/src/main/java/net/fabricmc/fabric/mixin/armor/MixinClass1183.java

```diff
@@ -0,0 +1,42 @@
+/*
+ * Copyright (c) 2016, 2017, 2018, 2019 FabricMC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package net.fabricmc.fabric.mixin.armor;
+
+import org.spongepowered.asm.mixin.Mixin;
+import org.spongepowered.asm.mixin.injection.At;
+import org.spongepowered.asm.mixin.injection.Redirect;
+
+import net.minecraft.class_1183;
+import net.minecraft.item.ArmorItem;
+import net.minecraft.item.Item;
+
+import net.fabricmc.api.EnvType;
+import net.fabricmc.api.Environment;
+import net.fabricmc.fabric.impl.armor.FabricArmorItem;
+
+@Environment(EnvType.CLIENT)
+@Mixin(class_1183.class)
+public class MixinClass1183 {
+	@Redirect(at = @At(value = "INVOKE", target = "Ljava/lang/String;format(Ljava/lang/String;[Ljava/lang/Object;)Ljava/lang/String;"), method = "method_4092")
```

Comment:
if there a way to not use redirect?

Reply:
the variable is hardcoded to take the name from the armor enum. `FabricArmorItem` does not use that enum, because it uses `ArmorMaterial`, hence it should take the name from `ArmorMaterial`(see getName()). Could also be done using a `ModifyVariable` or local capture at `INVOKE_ASSIGN`, but `Redirect` was easier.